ibis/tests/expr/test_operations.py

```diff
@@ -114,6 +114,37 @@ def output_type(self):
     assert isinstance(expr, MyTable)
 
 
+@pytest.mark.parametrize(
+    ['format_class', 'extension'],
+    [
+        (ops.ArrowFileFormat, 'arrow'),
+        (ops.OrcFileFormat, 'orc'),
+        (ops.ParquetFileFormat, 'parquet'),
+    ],
+)
+def test_file_table(format_class, extension):
+    node = ops.FileTable(
+        items=[
+            f's3://foo/2009/01/data.{extension}',
```

Comment:
We should expose this to the ibis api, e.g.:

```python
ibis.table(list_of_paths, schema)
```

though we may want to choose a different function name, not sure about the ergonomics.

cc @cpcloud

Reply:
I think `ibis.table` should remain the sole entry point for creating `UnboundTable`-based expressions. I'm not sure `FileTable`s should have their own distinct API beyond possibly getting them back from a call to `con.table(...)`.

Thoughts?

Reply:
I think there is an advantage of describing datasets on the operations level making them backend agnostic. How could we define and run queries agains a set of s3 files with different engines otherwise?

Reply:
Can you write down a small expression-creation-to-execution example? It will help drive the conversation.

Reply:
Since this is for a collection of files for a specific format, I'd assume this operation would be what's returned from an e.g. `read_parquet` call (if we were to add that). I think I agree with Phillip that `ibis.table` would feel a bit weird to handle this use case.

```python
table = con.read_parquet(list_of_paths_or_maybe_a_glob, **extra_options_here_perhaps)
```

Reply:
We do have some precedent for `con.parquet_file` in the impala backend and a few others like it `delimited_file`.

These APIs were never fully baked outside of the impala backend. Perhaps we should consider fully baking them, even if they turn out to look a bit different, for example different arguments or even making them plural (`parquet_files`, `text_files`, etc).

Reply:
I added a quick Pandas backend so you can do `ibis.pandas.parquet_files([...]).execute()`

Reply:
> Can you write down a small expression-creation-to-execution example? It will help drive the conversation.

Will do once I have the time.

Reply:
what if the schema is not consistent between files?

Reply:
@jayceslesar currently it would raise an error at runtime. I would expect the backends might deal with it in different ways: they could try to unify schemas or simply error.